tuxemon/states/pc/__init__.py

```diff
@@ -188,3 +203,158 @@ def initialize_items(self) -> Generator[MenuItem[None], None, None]:
+                )
+            )
+            new_image.scale(prepare.SCALE, prepare.SCALE)
+
+    def startup(self, box_name: String, **kwargs: Any) -> None:
+        width, height = prepare.SCREEN_SIZE
+
+        theme = get_theme()
+        theme.scrollarea_position = pygame_menu.locals.POSITION_EAST
+        columns = 3
+
+        self.box_name = box_name
+        self.player = local_session.player
+        self.box = self.player.monster_boxes[self.box_name]
+
+        num_mons = len(self.box)
+        # Widgets are like a pygame_menu label, image, etc.
+        num_widgets_per_monster = 2
+        rows = int(num_mons * num_widgets_per_monster / columns) + 1
+        # Make sure rows are divisible by num_widgets
+        while rows % num_widgets_per_monster != 0:
+            rows += 1
+
+        super().startup(
+            height=height, width=width, columns=columns, rows=rows, **kwargs
+        )
+
+        def take_monster(monster: monster) -> None:
+            self.player.remove_monster_from_storage(monster)
+            self.player.add_monster(monster)
+            open_dialog(
+                local_session,
+                [
+                    T.format(
+                        "menu_storage_take_monster", {"name": monster.name}
+                    )
+                ],
+            )
+            self.client.pop_state(self)
+
+        def take_monster_callback(monster: monster) -> Callable[[], object]:
+            return partial(take_monster, monster)
+
+        menu_items_map = []
+        for monster in self.box:
+            menu_items_map.append((monster, take_monster_callback(monster)))
+
+        self.add_menu_items(self.menu, menu_items_map)
+
+
+class MonsterBoxChooseState(PygameMenuState):
+    """Menu to choose a box, which you can then remove a tuxemon from."""
+
+    def add_menu_items(
+        self,
+        menu: pygame_menu.Menu,
+        items: Sequence[Tuple[str, MenuGameObj]],
+    ) -> None:
+
+        menu.add.vertical_fill()
+        for key, callback in items:
+            label = T.translate(key).upper()
+            menu.add.button(label, callback)
+            menu.add.vertical_fill()
+
+        width, height = prepare.SCREEN_SIZE
+        widgets_size = menu.get_size(widget=True)
+        b_width, b_height = menu.get_scrollarea().get_border_size()
+        menu.resize(
+            widgets_size[0],
+            height - 2 * b_height,
+            position=(width + b_width, b_height, False),
+        )
+
+    def startup(self, **kwargs: Any) -> None:
+        _, height = prepare.SCREEN_SIZE
+
+        super().startup(height=height, **kwargs)
+
+        self.animation_offset = 0
+
+        def change_state(state: str, **kwargs: Any) -> Callable[[], object]:
+            return partial(self.client.replace_state, state, **kwargs)
+
+        player = local_session.player
+        menu_items_map = []
+        for box_name in player.monster_boxes.keys():
```

Comment:
```
for box_name, monsters in player.monster_boxes.items():
    if not monsters:
        ...
```

Reply:
Iterating over the keys and then immediately getting the value from the dict is unidiomatic.
Using `len(some_iterable) == 0` rather than `not some_iterable` is unidiomatic.